x-pack/plugins/code/server/indexer/lsp_indexer.ts

```diff
@@ -58,6 +65,11 @@ export class LspIndexer extends AbstractIndexer {
     super.cancel();
   }
 
+  protected encodeCheckpoint(req: IndexRequest): string {
+    const r = req as LspIndexRequest;
+    return `${r.revision} - ${r.filePath}`;
```

Comment:
does it has to be a string?

a json seems more convient and then we could not call it encode

Reply:
we will persist this checkpoint into the ES, and thus need to provide a schema for this field. A json format would need a much complex schema, which is prone to future change.